MantidQt/SliceViewer/src/SliceViewer.cpp

```diff
@@ -670,6 +670,13 @@ void SliceViewer::updateDimensionSliceWidgets() {
     if (w > maxUnitsWidth)
       maxUnitsWidth = w;
 
+    // if the workspace is already binned, update the interface with the
+    // existing number of bins
+    if (boost::dynamic_pointer_cast<IMDHistoWorkspace>(m_ws)) {
```

Comment:
@samueljackson92 I think it might worth it. More `dynamic_cast`s only encourage further use of them. Thanks!